### ISSUE TITLE
World seems to crash Terramap if new (1.4.4.2) items are in chests

### ISSUE THREAD

**Issue by MimiaMia** [TerraMap/windows#41]
I am assuming that this is the cause; on the website, it loads from left to right up to where chests are contained with new items and then it goes crazy.  The application won't load at all.  Removing said items doesn't seem to fix it, but there might be an item somewhere I missed.  It worked until today when I put some of the new items in chests, the specific items I can think of are the vulkelf ears, Jojacola, the Star Drop (junimo minion), and _maybe_ one of the paintings.

(The only other thing worth mentioning at all is that I did mining near the bottom of the map directly where the map starts to screw up, but it doesn't line up directly.)

This is what the map looks like on the website:
![image](https://user-images.githubusercontent.com/114980616/193755274-c9ce84dd-4348-4259-980c-4e6e5e00fae7.png)
![image](https://user-images.githubusercontent.com/114980616/193757042-e8f66e59-90e2-4782-aff4-6573b980a57b.png)

As you can see it loads all the way up to where my chests appear (you can see the very left part of the house I built loading ok) and then it just totally breaks the map.  It loads just fine in Terraria so I don't think anything is wrong with the map.

Here are my error codes when I tried to load the map on the application:
(note 1: It gave two exception errors so I posted both of them here)
(note 2: I removed the user folder from it manually, though it doesn't seem to reference my own C drive folder anyways)

System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at TerraMap.Data.World.ReadTilesVersion2(BinaryReader reader, Boolean[] importance) in C:\Users\~\source\terramap\windows\Data\World.cs:line 1097
   at TerraMap.Data.World.ReadWorldVersion2(BinaryReader reader) in C:\Users\~\source\terramap\windows\Data\World.cs:line 483
   at TerraMap.Data.World.Read(String filename) in C:\Users\~\source\terramap\windows\Data\World.cs:line 456
   at TerraMap.Data.World.<>c__DisplayClass406_0.<ReadAsync>b__0() in C:\Users\~\source\terramap\windows\Data\World.cs:line 418
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at TerraMap.MainWindow.<Open>d__24.MoveNext() in C:\Users\~\source\terramap\windows\TerraMap\MainWindow.xaml.cs:line 308
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at TerraMap.MainWindow.<Open>d__23.MoveNext() in C:\Users\~\source\terramap\windows\TerraMap\MainWindow.xaml.cs:line 275
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at TerraMap.MainWindow.<OnOpenWorldFile>d__59.MoveNext() in C:\Users\~\source\terramap\windows\TerraMap\MainWindow.xaml.cs:line 1244
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at TerraMap.Data.World.ReadTilesVersion2(BinaryReader reader, Boolean[] importance) in C:\Users\~\source\terramap\windows\Data\World.cs:line 1097
   at TerraMap.Data.World.ReadWorldVersion2(BinaryReader reader) in C:\Users\~\source\terramap\windows\Data\World.cs:line 483
   at TerraMap.Data.World.Read(String filename) in C:\Users\~\source\terramap\windows\Data\World.cs:line 456
   at TerraMap.Data.World.<>c__DisplayClass406_0.<ReadAsync>b__0() in C:\Users\~\source\terramap\windows\Data\World.cs:line 418
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at TerraMap.MainWindow.<Open>d__24.MoveNext() in C:\Users\~\source\terramap\windows\TerraMap\MainWindow.xaml.cs:line 308
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at TerraMap.MainWindow.<Open>d__23.MoveNext() in C:\Users\~\source\terramap\windows\TerraMap\MainWindow.xaml.cs:line 275
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at TerraMap.MainWindow.<OnSelectedWorldFileChanged>d__61.MoveNext() in C:\Users\~\source\terramap\windows\TerraMap\MainWindow.xaml.cs:line 1266
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

You might already be aware of these problems and I am sure you will be updating Terramap soon for full 1.4.4.2 support regardless; thank you for all your hard work!

**Comment by MimiaMia** [comment]
I did a little further testing.  

When I went to a new world, I put in the new items and the map broke, however when I used the website it did not have the map tearing apart like it did previously.  When I removed the items, it loaded normally.

I also had used illuminant paint (newly added in 1.4.4.2) and it also breaks the file and prevents it from loading.  Again, removing the paint let it work again.  In both cases the map still loaded normally on the website, but I had only used about 23 tiles of paint on the background wall and tiles (standard wooden wall and wood blocks).

I cannot confirm why the original map loaded in such a broken way.  However, since I used illuminant paint, I want to mention that I specifically used it on grey stone bricks, grey stone brick walls, and grey stone brick platforms.
